task.js

```diff
@@ -0,0 +1,231 @@
+    return this.marks.reduce((a, b) => (a + b)) / this.marks.length;
+  }
+
+  // returns the lowest score
+  getMinMark() {
+    return Math.min(...this.marks)
+  }
+
+  // returns the maximum score
+  getMaxMark() {
+    return Math.max(...this.marks)
+  }
+}
+
+// ------------------------------------Teacher-------------------------------
+
+class Teacher extends Human{
+  constructor(info) {
+    super(info);
+    this.students =  info.students
+
+    //Creates an object where the keys are the Name and the value is an array of objects
+    // this.students = students.reduce((group, arr) => {
+    //   const { name } = arr;
+    //   group[name] = group[name] ?? [];
+    //   group[name].push(arr);
+    //   return group;
+    // }, {}); 
+  } 
+
+  // returns an array of student names sorted by grade
+  getListOfNamesByAverageMark() {  
+    return [...this.students].sort((x, y) => x.getAverageMark() - y.getAverageMark()).map((element) => (element.name ) )
+  }
+
+  // return object  student from name 
+  getStudentByName(name) {   
+    return this.students.find(student => student.name == name)  
+  }
+
+  // change or delete from name 
+  removeAndUpdate(name = false, newObject = false) { 
+    let index = this.students.findIndex( (element) => element.name == name) 
+    if (index != -1) {  
+      if (newObject) {
+        this.students.splice(index, 1,newObject)
+      }
+      else {
+        this.students.splice(index, 1)
+      }
+    }
+    else {
+      console.log("There is no such name ------ ")
+    }
+  }
+
+  // replace a student with another one knowing his or her name
+  updateStudentByName(newStudent, name) {
+    this.removeAndUpdate(name, newStudent)
+  }
+  // delete object student from name  
+  removeStudentByName(name) {
+    this.removeAndUpdate(name)
+  }
+
+  // prints a string with the real grades of the fake students
+  findFakeStudent(){
+    let result = ''
+    // for (let iName = 0; iName < this.students.length; iName++) {   
+    //   let averagMarks =  this.students[iName].marks.reduce((a, b) => (a + b)) / this.students[iName].marks.length 
+    //   if (averagMarks != this.students[iName].getAverageMark()) 
+    //     result += `${this.students[iName].name} - ${this.students[iName].marks} \n`
+    // }
+
+    // the second option that focuses on the class name  "FakeStudent"
+    for (let iName = 0; iName < this.students.length; iName++) {  
+      if (this.students[iName] instanceof FakeStudent) 
+        result += `${this.students[iName].name} - ${this.students[iName].marks} \n`
+    }  
+    console.log(result)
+  }
+}
+
+// --------------------------------------FakeStudent-------------------------
+
+class FakeStudent extends Student {
+  #chetedMarks =  this.#cheat();
+  
+  #cheat() {
+    return this.marks.map((x) => ((x *= 2) < 13) ? x : 12 )
+  }
+
+  // returns the average score
+  getAverageMark() {
+    return this.#chetedMarks.reduce((a, b) => (a + b)) / this.#chetedMarks.length;
```

Comment:
Create a static method to not repeat your code and use it for `getAverageMark` at `Student` and for this case